### DISCUSSION TITLE
Issue with converting to nwb files

### DISCUSSION THREAD

**Discussion by timheistek** [AllenInstitute/MIES#1852] [Q&A] [answered]
When converting to nwb files we sometimes get the following error:

 !!! Threadsafe assertion FAILED !!!
  Message: "Operation IPNWB_WriteCompound not present."
  Please provide the following information if you contact the MIES developers:
  ################################
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  Stacktrace:
NWB_ExportWithDialog(...)#L834 [MIES_NeuroDataWithoutBorders.ipf]
NWB_ExportAllData(...)#L646 [MIES_NeuroDataWithoutBorders.ipf]
NWB_AppendSweepLowLevel(...)#L1126 [MIES_NeuroDataWithoutBorders.ipf]
WriteSingleChannel(...)#L516 [IPNWB_Writer.ipf]
WriteEpochs(...)#L535 [IPNWB_Writer.ipf]
AppendToEpochTable(...)#L609 [IPNWB_Writer.ipf]
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  Time: 2023-08-15T11:05:32+02:00
  Experiment: 20230713_31_02 (Packed)
  Igor Pro version: 9.0.1.2 (39200)
  ################################

Could you help with this?

**Comment by t-b** [comment]
Thanks for the report.

The operation `IPNWB_WriteCompound` is implemented in the mies-nwb2-compound-XOP-64 XOP. So I would expect that it either always works or never.

- How have you installed? Installer or another method?
- If installer with or without hardware XOPs?
- What does `Mies panels->Check Installation` return?

**Comment by timheistek** [comment]
MIES is installed with installer, without hardware XOPs. Also on the PC we use for recording there is an error with converting these files into nwb. This is the return of check installation:

  Checking system properties:
  Igor 64bit: 9.01, 39200
  Windows 10: True
  ASLR: True (Very Bad)
  
Checking known defines:
  DEBUGGING_ENABLED: No
  EVIL_KITTEN_EATING_MODE: No
  BACKGROUND_TASK_DEBUGGING: No
  THREADING_DISABLED: No
  
Checking base installation:
  Mies version info: Valid "Release_2.6_20220606-722-g9741f982b..." (Nice!)
  ITC XOP: The file itcxop2-64.xop could not be found (Very Bad)
  JSON XOP: Found version 1.0.0.1 (Nice!)
  VDT2 XOP: The file VDT2-64.xop could not be found (Very Bad)
  Axon Telegraph XOP: The file AxonTelegraph64.xop could not be found (Very Bad)
  Multi Clamp Commander XOP: The file MultiClamp700xCommander64.xop could not be found (Very Bad)
  ZeroMQ XOP: Found version 1.0.0.1 (Nice!)
  TUF XOP: Found version 1.0.0.1 (Nice!)
  JSON XOP: Present in the right version (version-817-g0ceb12c) (Nice!)
  ITC2 XOP: Present in the wrong version (expected: latest-170-g1dfb5c5 vs present: error querying version) (Very Bad)
  TUF XOP: Present in the right version (version-163-g686effb) (Nice!)
  Results: 9 checks, 6 number of errors
  
Checking extended installation:
  NI-DAQ MX XOP: The file NIDAQmx64.xop could not be found (Very Bad)
  Results: 1 checks, 1 number of errors

**Comment by t-b** [comment]
Thanks for the check installation output.

I think we have two different bugs here.

1.) The reason that the NWB export does not work with an installation without hardware is fixed in #1853. You can get the installer at https://github.com/AllenInstitute/MIES/actions/runs/5929963671?pr=1853 you need the 
"BuildInstaller-user-assets" artefact.

Can you try that?

2.) I don't know why the NWB export would sometimes fails on a full installation. If that still happens with the installer from 1.) I would need the pxp. I can provide FTP details for upload if required.

**Comment by timheistek** [comment, marked as answer]
Thanks for the update! On the full installation the NWB export works fine. I haven't tried on a pc without hardware.

**Comment by t-b** [comment]
Closing for now, please reopen if there is still something to fix.
